net/lp_rpc.proto

```diff
@@ -20,12 +20,62 @@ message TranscoderRequest {
   bytes sig   = 2;
 }
 
+/*
+  OSInfo needed to negotiate storages that will be used.
+  It carries info needed to write to the storage.
+*/
+message OSInfo {
+
+  // Storage type: s3, ipfs.
+  string storage = 1;
+
+  S3OSInfo s3info = 16;
+}
+
+message S3OSInfo {
+
+  // Configured S3 bucket name
+  string bucket = 1;
+
+  // Configured S3 region
+  string region = 2;
+ 
+  // POST policy that S3 owner node creates to give write access to other node.
+  string policy = 3;
+
+  // Signature for POST policy.
+  string signature = 4;
+
+  // Needed for POST policy.
+  string xAmzCredential = 5;
+
+  // Needed for POST policy.
+  string xAmzDate = 6;
+}
+
+message SessionInfo {
+  // nonce - unique identificator of video stream, used in folder name there data saved in storage
+  // (folder name matters when transcoder puts trasncoded data into B's owned storage).
+  uint64 nonce = 1;
+
+  OSInfo osInfo = 2;
+}
+
+// Used in POST /segment call to carry info about segment to transcode from broadcaster to transcoder
+// Can contain credentials of OS where broadcaster wants transcoded data to be put.
+message SegmentInfo {
+
+  TypedURI turi = 1;
+
+  SessionInfo sessionInfo = 2;
```

Comment:
I'll throw out nonce, so it will be
```
Segment Info
  Typed URI
  OSInfo
```
maybe it is not clear from that structure, but Typed URI points to input segment, but OSInfo contains info about OS for output segments to be put to.

Reply:
Following https://github.com/livepeer/go-livepeer/issues/582 : what do you think about putting `OSInfo` into the `SegData` protobuf structure? (And maybe `repeated` as well).

That allows us to submit the OSInfo separately from the body. Then the body can be a URI, or directly uploaded raw video data.

How big is a typical `S3Creds` data? If it's under a few KB then it's probably fine to stuff in the HTTP headers alongside `SegData`.

Reply:
I've checked - OSInfo with S3 policy encoded is around 900 characters (encoded into base64)
I'll move OSInfo into `SegData`